### PR TITLE
Update module standards/tests/scripts

### PR DESCRIPTION
**Pull request by xdumaine**
What:

- Convert module to es6
  - no more self/var
  - class/extends
- add basic test framework
  - Basically just show that tests work in creating an instance of the module
- add building to es5 for consumption
  - Adapter.js does this, which is handy for most projects that require() in modules
- add linting

Why:

- Writing code in this module is tedious because of lack of es6+ support. It's messy and just looks bad with `var self = this` all over the place. 
- It needs tests. This is in use in production applications and should be tested at the unit level at least. Having code that's easier to write and work with makes writing tests easier.
- These things all encourage project growth/stability/community participation

TODO:
- [ ] Get buy-in from interested parties on the approaches I've taken here
  - [x] Switch from Tape to mocha/sinon
- [ ] Expand on tests to have a baseline
- [ ] Once we've settled on approaches/styles/etc, apply these to the other related modules

### CODE REVIEW

**Comment by kenny-house** [comment]
Would there be interest in working on a mock RTCPeerConnection (the browser API, not the node module) for some of this testing, or potentially to be integrated later? There may already be a good mock out there I'm just not familiar with. I could see it being useful for testing the functions with async sequences in jingle-media-session.

**Comment by xdumaine** [comment]
At my company, we make use of a fork of zombese with additional webrtc mocks. It serves it's purpose fairly well. Definitely for the unit tests in this module a dead simple object mock is sufficient. For the future and integration testing, I think zombese or just a headless chrome with true WebRTC support is the way to go. 

Https://GitHub.com/mypurecloud/zombese

File: package.json [hunk 1]
```diff
@@ -4,19 +4,22 @@
   "author": "Lance Stout <lance@andyet.net>",
   "bugs": "https://github.com/otalk/jingle-media-session/issues",
   "contributors": [
-    "Philipp Hancke <fippo@andyet.net>"
+    "Philipp Hancke <fippo@goodadvicepages.net>",
```
Comment (fippo) [inline package.json]: just remove the email please. This one is wrong but the real one gets enough spam as is

File: test/test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const test = require('tape');
```
Comment (fippo) [inline test/test.js]: i am fairly happy with mocha these days, in particular together with sinon.

**Comment by fippo** [comment]
No good RTCPeerConnection mocks unfortunately. I've cursed the lack of this many times...

i'm no longer using this in production (sadly) so deferring to Lance

File: test/test.js (hunk 2 above)
Comment (xdumaine) [reply test/test.js]: I tend to use mocha and sinon as well. I'll switch to that. Used tape as it was already a dependency, but no tests...

**Comment by xdumaine** [comment]
@legastero thoughts?

**Comment by xdumaine** [comment]
@legastero did you you have any opinions on this? If you'd rather not go through the trouble, that's cool, just let me know.

**Pull request closed without merging**